buildSrc/src/main/java/net/neoforged/neodev/e2e/RunProductionClient.java

```diff
@@ -0,0 +1,362 @@
+     * The NeoForge version, used for placeholders when launching the game.
+     * It needs to match the installer used.
+     */
+    @Input
+    public abstract Property<String> getNeoForgeVersion();
+
+    /**
+     * The original, unmodified client jar.
+     * The Vanilla launcher puts this on the classpath when it launches the game.
+     */
+    @InputFile
+    public abstract RegularFileProperty getOriginalClientJar();
+
+    @Inject
+    public RunProductionClient(ExecOperations execOperations) {
+        this.execOperations = execOperations;
+    }
+
+    @TaskAction
+    @Override
+    public void exec() {
+        var installDir = getInstallationDir().getAsFile().get().toPath();
+        var nativesDir = installDir.resolve("natives");
+        try {
+            Files.createDirectories(nativesDir);
+        } catch (IOException e) {
+            throw new GradleException("Failed to pre-create natives directory " + nativesDir, e);
+        }
+        var librariesDir = installDir.resolve("libraries");
+
+        var minecraftVersion = getMinecraftVersion().get();
+        var versionId = "neoforge-" + getNeoForgeVersion().get();
+
+        var assetProperties = new Properties();
+        try (var in = new FileInputStream(getAssetPropertiesFile().getAsFile().get())) {
+            assetProperties.load(in);
+        } catch (IOException e) {
+            throw new GradleException("Failed to read asset properties " + getAssetPropertiesFile(), e);
+        }
+
+        var assetIndex = Objects.requireNonNull(assetProperties.getProperty("asset_index"), "asset_index");
+        var assetsRoot = Objects.requireNonNull(assetProperties.getProperty("assets_root"), "assets_root");
+
+        // Set up the placeholders generally used by Vanilla profiles in their argument definitions.
+        var placeholders = new HashMap<String, String>();
+        placeholders.put("auth_player_name", "Dev");
+        placeholders.put("version_name", minecraftVersion);
+        placeholders.put("game_directory", installDir.toAbsolutePath().toString());
+        placeholders.put("auth_uuid", "00000000-0000-4000-8000-000000000000");
+        placeholders.put("auth_access_token", "0");
+        placeholders.put("clientid", "0");
+        placeholders.put("auth_xuid", "0");
+        placeholders.put("user_type", "legacy");
+        placeholders.put("version_type", "release");
+        placeholders.put("assets_index_name", assetIndex);
+        placeholders.put("assets_root", assetsRoot);
+        placeholders.put("launcher_name", "NeoForgeProdInstallation");
+        placeholders.put("launcher_version", "1.0");
+        placeholders.put("natives_directory", nativesDir.toAbsolutePath().toString());
+        // These are used by NF but provided by the launcher
+        placeholders.put("library_directory", librariesDir.toAbsolutePath().toString());
+        placeholders.put("classpath_separator", File.pathSeparator);
+
+        execOperations.javaexec(spec -> {
```

Comment:
Why are you working with execOperations instead of using the `JavaExec` machinery?

Reply:
Ah you think it might be adding random things to all `JavaExec` tasks?

Reply:
Alright